src/events/tls.py

```diff
@@ -111,7 +111,14 @@ def _on_certificate_available(self, event: CertificateAvailableEvent) -> None:
             logger.error("Can't use certificate, found unknown CSR")
             return
 
-        self.charm.state.unit_server.update({"certificate": event.certificate, "ca": event.ca})
+        # if certificate already exists, this event must be new, flag manual restart
+        if self.charm.state.unit_server.certificate:
+            self.charm.on[f"{self.charm.restart.name}"].acquire_lock.emit()
+
+        self.charm.state.unit_server.update(
+            {"certificate": event.certificate, "ca-cert": event.ca}
+        )
+        self._cleanup_old_ca_field()
```

Comment:
**question/suggestion:** I might be misunderstanding the plan here with both keys being available, so if I'm wrong please let me know. 

On this new event, we're re-requesting a cert, and saving that 'new' one to `ca-cert` and cleaning up the old one.

If that's understood, I don't think we need the restart here.
I also think we can do just this:

```python
self.charm.state.unit_broker.update(
    {"certificate": event.certificate, "ca-cert": event.ca, "ca": ""}
)
```
and not need the `cleanup_old_ca_field` function.

Reply:
Note https://github.com/canonical/zookeeper-operator/pull/129#discussion_r1570444080

Reply:
Done.